x-pack/solutions/security/plugins/security_solution/server/lib/siem_migrations/dashboards/data/dashboard_migrations_data_resources_client.ts

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { sha256 } from 'js-sha256';
+import type { QueryDslQueryContainer, Duration } from '@elastic/elasticsearch/lib/api/types';
+
+import type {
+  SiemMigrationResource,
+  SiemMigrationResourceBase,
+  SiemMigrationResourceType,
+} from '../../../../../common/siem_migrations/model/common.gen';
+import { SiemMigrationsDataBaseClient } from '../../common/data/siem_migrations_data_base_client';
+
+export type CreateDashboardMigrationResourceInput = Pick<
+  SiemMigrationResource,
+  'migration_id' | 'type' | 'name' | 'content' | 'metadata'
+>;
+
+export interface DashboardMigrationResourceFilters {
+  type?: SiemMigrationResourceType;
+  names?: string[];
+  hasContent?: boolean;
+}
+
+/* BULK_MAX_SIZE defines the number to break down the bulk operations by.
+ * The 500 number was chosen as a reasonable number to avoid large payloads. It can be adjusted if needed.
+ */
+const BULK_MAX_SIZE = 500 as const;
+/* DEFAULT_SEARCH_BATCH_SIZE defines the default number of documents to retrieve per search operation
+ * when retrieving search results in batches. */
+const DEFAULT_SEARCH_BATCH_SIZE = 500 as const;
+
+export class DashboardMigrationsDataResourcesClient extends SiemMigrationsDataBaseClient {
```

Comment:
This is the same exact class implementation as the _RuleMigrationsDataResourcesClient_, with different names. We should consider using a unified class on both sides (like _SiemMigrationsDataResourcesClient_ or just _MigrationsDataResourcesClient_)